ui/src/pages/namespace/Explorer/Page/index.tsx

```diff
@@ -0,0 +1,53 @@
+import { Card } from "~/design/Card";
+import { FC } from "react";
+import { NoPermissions } from "~/design/Table";
+import PageEditor from "./PageEditor";
+import { PanelTop } from "lucide-react";
+import { analyzePath } from "~/util/router/utils";
+import { useFile } from "~/api/files/query/file";
+import { useNamespace } from "~/util/store/namespace";
+import { usePages } from "~/util/router/pages";
+
+const UIPage: FC = () => {
+  const pages = usePages();
+  const { path } = pages.explorer.useParams();
+  const namespace = useNamespace();
+  const { segments } = analyzePath(path);
+  const filename = segments[segments.length - 1];
+
+  const {
+    isAllowed,
+    noPermissionMessage,
+    data: endpointData,
+    isFetched: isPermissionCheckFetched,
+  } = useFile({ path });
+
+  if (!namespace) return null;
+  if (!path) return null;
+  if (endpointData?.type !== "endpoint") return null;
```

Comment:
As I'm reviewing this, there has been an update in the backend from yassir that happened after your last message. I  wiped my local backend to make this changes apply and after that I had to change `endpoint` to `page` to make this page work. 

But with this change, the breadcrumb and the file icon in the file explorer automatically worked. 🤩